src/validation.h

```diff
@@ -639,7 +640,8 @@ class CChainState
      */
     bool ActivateBestChain(
         BlockValidationState& state,
-        std::shared_ptr<const CBlock> pblock = nullptr) LOCKS_EXCLUDED(m_chainstate_mutex, cs_main);
+        std::shared_ptr<const CBlock> pblock = nullptr)
+        EXCLUSIVE_LOCKS_REQUIRED(!::cs_main, !m_chainstate_mutex);
```

Comment:
Tend toward NACK using negative annotations with global mutexes, especially cs_main. I presume `LOCKS_EXCLUDED(::cs_main) EXCLUSIVE_LOCKS_REQUIRED(!m_chainstate_mutex)` doesn't work?

Reply:
What's wrong with using negative annotations with global mutexes? I don't see how the scope of the mutex is relevant, but maybe I miss something since @hebasto also wasn't excited about it. Can you elaborate?

> I presume `LOCKS_EXCLUDED(::cs_main) EXCLUSIVE_LOCKS_REQUIRED(!m_chainstate_mutex)` doesn't work?

It works. Updated the PR.

Reply:
> Can you elaborate?

See the commit that you ACKed: 9ac8f6d7ddc4c949d84de5584baa813144c7fb26

* It is wasted effort, since cs_main is recursive
* It is impossible to properly implement: https://github.com/bitcoin/bitcoin/pull/20272#discussion_r515215137
* It is too verbose for globals, as they propagate outside the scope of "their" `class`